tools/update_init_file.py

```diff
@@ -92,7 +92,7 @@ def relevant_attributes(namespace: dict[str, Any], /) -> list[str]:
     it = (
         name
         for name, attr in namespace.items()
-        if (not name.startswith("_")) and _is_relevant(attr)
+        if (not name.startswith("_")) and _is_relevant(attr, name)
```

Comment:
@binste this could also work, without re-introducing `name` to the attribute checking

```py
it = (
	name
	for name, attr in namespace.items()
	if (not name.startswith(("_", "pd", "jsonschema")) and _is_relevant(attr)
)
```

Another option could be adding a global tuple of banned names, like with `_TYPING_CONSTRUCTS`.

What do you think?

Reply:
I don't see a downside of introducing name as an argument and it's more explicit to me if the function handles all checks if something is relevant. startswith is a bit obscure for a new code reader as we don't expect multiple entries to start with "pd".

As it's only used once, I don't think we need a global constant but no real preference.

Reply:
> As it's only used once, I don't think we need a global constant but no real preference

@binste that's fair, with how few names there are now, I'm fine with it for now! 

For context, I'm trying to avoid slipping back into the earlier version:
https://github.com/vega/altair/blob/18a2c3c237014591d172284560546a2f0ac1a883/tools/update_init_file.py#L86-L119